modules/uac_registrant/registrant.c

```diff
@@ -482,7 +482,7 @@ int run_reg_tm_cback(void *e_data, void *data, void *r_data)
 		while (contact) {
 			/* Check for binding */
 			if (contact->uri.len==rec->contact_uri.len &&
-				strncmp(contact->uri.s,rec->contact_uri.s,contact->uri.len)==0){
+				strncasecmp(contact->uri.s,rec->contact_uri.s,contact->uri.len)==0){
```

Comment:
This is not correct because if the username was altered, the URIs are not the same:
 `sip:alice@AtLanTa.CoM;Transport=udp`             (different usernames)
 `sip:ALICE@AtLanTa.CoM;Transport=UDP`

Reply:
Thanks, what would you think of adding a `strncmp_lc_transport` that takes a `pkg_malloc` copy of both strings, turns the `;transport=...` into lowercase, then `strcmp`s the copies with lowercased transport, and then frees the copies and returns the result of the `strcmp`?

Reply:
Take a look at `compare_uris()` from `parse_uri.c`.
It does the proper job if URIs are passed as parsed structures, but comparing raw URIs is bad :(
I pushed some fixes on the `compare_uris` branch:
https://github.com/OpenSIPS/opensips/tree/compare_uris
Please give it a try and let me know the results.

Reply:
Did you test the new branch in production and it's working ok for you?

Reply:
Not in production, no, only in a test setup. I registered a user with incorrect case in the `transport` param and it worked & did not cause any obvious issues.